accounts/utils.py

```diff
@@ -0,0 +1,80 @@
+from datetime import timedelta
+
+import requests
+from django.utils import timezone
+
+from accounts.settings import accounts_settings
+
+
+def authenticated_request(user, method, url,
+                          params=None, data=None, headers=None, cookies=None, files=None,
+                          auth=None, timeout=None, allow_redirects=True, proxies=None,
+                          hooks=None, stream=None, verify=None, cert=None, json=None):
+    """
+    Helper method to make an authenticated request using the user's access token
+    NOTE be ABSOLUTELY sure you only make a request to Penn Labs products, otherwise
+    you will expose user's access tokens to the URL you provide and bad things will
+    happen
+    """
+
+    # Access token is expired. Try to refresh access token
+    if user.accesstoken.expires_at < timezone.now():
+        if not refresh_access_token(user):
+            return None  # Couldn't update access token
+
+    # Update Headers
+    headers = headers or {}
+    headers['Authorization'] = f'Bearer {user.accesstoken.token}'
+
+    # Make request
+    s = requests.Session()
```

Comment:
maybe explain why you're using a session here?

Reply:
Honestly, I'm just using the session because it makes it easier to make a request where you can define the method (`GET`, `POST`, etc). Using the base requests module would require somehow calling `.get()` or  `.post()`, which would be more work than I think is necessary

Reply:
Think it'd be good to have some explanation then

Reply:
I can definitely add a comment in the code if that's what you mean, but I don't think it's worth adding to the README